lib/tbot/service_identity_output.go

```diff
@@ -374,6 +377,54 @@ func renderSSHConfig(
 		}); err != nil {
 			return trace.Wrap(err)
 		}
+
+		// Generate the per cluster files
+		for _, clusterName := range clusterNames {
+			sshConfigName := fmt.Sprintf("%s.%s", clusterName, ssh.ConfigName)
+			knownHostsName := fmt.Sprintf("%s.%s", clusterName, ssh.KnownHostsName)
+			knownHostsPath := filepath.Join(absDestPath, knownHostsName)
+
+			sb := &strings.Builder{}
+			if err := sshConf.GetClusterSSHConfig(sb, &openssh.ClusterSSHConfigParameters{
+				AppName:             openssh.TbotApp,
+				ClusterName:         clusterName,
+				KnownHostsPath:      knownHostsPath,
+				IdentityFilePath:    identityFilePath,
+				CertificateFilePath: certificateFilePath,
+				ProxyHost:           proxyHost,
+				ProxyPort:           proxyPort,
+				ExecutablePath:      executablePath,
+				DestinationDir:      absDestPath,
+
+				Insecure:          botCfg.Insecure,
+				FIPS:              botCfg.FIPS,
+				TLSRouting:        proxyPing.Proxy.TLSRoutingEnabled,
+				ConnectionUpgrade: connUpgradeRequired,
+				// Session resumption is enabled by default, this can be
+				// configurable at a later date if we discover reasons for this to
+				// be disabled.
+				Resume: true,
+			}); err != nil {
+				return trace.Wrap(err)
+			}
+			if err := destDirectory.Write(ctx, sshConfigName, []byte(sb.String())); err != nil {
+				return trace.Wrap(err)
+			}
+
+			knownHosts, ok := clusterKnownHosts[clusterName]
+			if !ok {
+				log.DebugContext(
+					ctx,
+					"No generated known_hosts for cluster, will skip",
+					"cluster", clusterName,
+				)
+				continue
```

Comment:
Hmm, is this something that's likely to happen in normal use? I'd almost be inclined to make it warn level if not.

Though realistically, we'll be able to ask for debug logs if we ever get a ticket asking why a cluster is missing, so maybe it doesn't really matter.

Reply:
[9fdd382](https://github.com/gravitational/teleport/pull/46397/commits/9fdd382d767b9b322d83fff7d3ceef46efb785f3)